app/controllers/users_controller.rb

```diff
@@ -15,17 +15,25 @@ def edit
   def create
     authorize User, :create?
     @user = User.new(permitted_attributes(User.new))
+    if @user.save
+      UserMailer.registration_confirmation(@user).deliver
+      redirect_to new_user_sessions_path, notice: 'User was successfully created. Please confirm your email.' 
+    else
+      render action: 'new' 
+    end
+  end
 
-    respond_to do |format|
-      if @user.save
-        cookies.permanent[:user_id] = @user.id
-        format.html { redirect_to root_path, notice: 'User was successfully created.' }
-        format.json { render action: 'show', status: :created, location: @user }
-      else
-        format.html { render action: 'new' }
-        format.json { render json: @user.errors, status: :unprocessable_entity }
-      end
+  def confirm_email
+    @user = User.find_by_id(params[:id])
+    if @user && @user.activation_token == params[:token]
```

Comment:
Wyświetli się wtedy komunikat "The activation link has already been used or is invalid. Please try to log in." i user zostanie przekierowany na stronę logowania. Czy to źle? :P